Parse concrete private interface methods in Java

Java 9+ allows concrete private interface methods, but the scalac parser doesn't accept them

```
$> scalac B.java
B.java:2: error: `;` expected but `{` found.
  private void foo() { return; }
                     ^
1 error
```

--- a/src/compiler/scala/tools/nsc/javac/JavaParsers.scala
+++ b/src/compiler/scala/tools/nsc/javac/JavaParsers.scala
@@ -613,7 +613,7 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
           val vparams = formalParams()
           if (!isVoid) rtpt = optArrayBrackets(rtpt)
           optThrows()
-          val isConcreteInterfaceMethod = !inInterface || (mods hasFlag Flags.JAVA_DEFAULTMETHOD) || (mods hasFlag Flags.STATIC)
+          val isConcreteInterfaceMethod = !inInterface || (mods hasFlag Flags.JAVA_DEFAULTMETHOD) || (mods hasFlag Flags.STATIC) || (mods hasFlag Flags.PRIVATE)
           val bodyOk = !(mods1 hasFlag Flags.DEFERRED) && isConcreteInterfaceMethod
           val body =
             if (bodyOk && in.token == LBRACE) {

--- a/test/files/pos/t12393/R1.java
+++ b/test/files/pos/t12393/R1.java
@@ -1,0 +1,7 @@
+// javaVersion: 9+
+public interface R1 {
+
+    private void foo() {
+        return;
+    }
+}

